conman/routes/managers.py

```diff
@@ -30,3 +32,31 @@ def best_match_for_path(self, path):
         except IndexError:
             msg = 'No matching Route for URL. (Have you made a root Route?)'
             raise self.model.DoesNotExist(msg)
+
+    def move_branch(self, old_url, new_url):
+        """
+        Move this Route and all descendants to a new URL.
+
+        Movement will be atomic. This means that failure to move any one Route
+        will cause all movement to fail. A conflicting URL will cause an
+        IntegrityError.
+        """
+        to_move = self.filter(url__startswith=old_url)
+        to_move.update(url=Concat(
+            Value(new_url),
+            Substr('url', len(old_url) + 1),  # 1 indexed
+        ))
+
+    def swap_urls(self, path_a, path_b):
+        """
+        Swap the urls of two Routes without moving their descendants.
```

Comment:
Further thoughts:

> This wouldn't significantly change the code here (if I understand correctly)

I'm beginning to think that the `update` method employed here is not a good idea, and that I should instead be doing a `SELECT` to get the `Routes` followed by a `save()`. ([I've alluded to this in another comment.](https://github.com/meshy/django-conman/pull/80#discussion_r69424057)) This would allow two things:

- We could raise errors when asked to swap things that do not exist.
- We might add a hook for custom logic/validation.

This second point springs from an idea I have for allowing `Routes` to have  dominion over sub-URLs. ie: the ArticleRoute at `/articles/` would be invoked when the `/articles/2016-07-04-flag-waving` url is hit, and, as such, might dictate that having child urls is disallowed.

Disallowing children might well make this significantly more complex, but is a concept I hope to introduce (unsure if it should be core...probably though).